head/src/main/kotlin/io/qalipsis/core/head/report/thymeleaf/SVGReplacedElementFactory.kt

```diff
@@ -0,0 +1,61 @@
+package io.qalipsis.core.head.report.thymeleaf
+
+import com.lowagie.text.Image
+import org.apache.commons.io.IOUtils
+import org.w3c.dom.Element
+import org.xhtmlrenderer.extend.FSImage
+import org.xhtmlrenderer.extend.ReplacedElement
+import org.xhtmlrenderer.extend.ReplacedElementFactory
+import org.xhtmlrenderer.extend.UserAgentCallback
+import org.xhtmlrenderer.layout.LayoutContext
+import org.xhtmlrenderer.pdf.ITextFSImage
+import org.xhtmlrenderer.pdf.ITextImageElement
+import org.xhtmlrenderer.render.BlockBox
+import org.xhtmlrenderer.simple.extend.FormSubmissionListener
+import java.io.FileInputStream
+
+
+/**
+ * Custom implementation of [ReplacedElementFactory] that defines the rendering process for image elements.
+ *
+ * @author Francisca Eze
+ */
+internal class SVGReplacedElementFactory(private val superFactory: ReplacedElementFactory) : ReplacedElementFactory {
+
+    override fun createReplacedElement(
+        layoutContext: LayoutContext?, blockBox: BlockBox,
+        userAgentCallback: UserAgentCallback?, cssWidth: Int, cssHeight: Int
+    ): ReplacedElement? {
+        val element: Element = blockBox.element
+        val nodeName = element.nodeName
+        // Replace any <img class="img" /> with the binary data of `image.svg` into the PDF.
+        if ("img" == nodeName) {
+            //Returns the image src.
+            val imagePath: String = element.getAttribute("src")
+            try {
+                val inputStream = FileInputStream(imagePath)
+                inputStream.use {
+                    val byteArray = IOUtils.toByteArray(inputStream)
+                    val image = Image.getInstance(byteArray)
+                    val fsImage: FSImage = ITextFSImage(image)
+                    if (cssWidth != -1 || cssHeight != -1) {
+                        fsImage.scale(cssWidth, cssHeight)
+                    } else {
+                        fsImage.scale(2000, 1000)
+                    }
+                    return ITextImageElement(fsImage)
+                }
+            } catch (e: Exception) {
+                throw RuntimeException("There was a problem trying to read a template embedded graphic: $e")
```

Comment:
Perhaps you could create a specific exception type instead of using RuntimeException, which is not the best choice...